app/api/public-cloud/aws-roles/helpers.ts

```diff
@@ -40,6 +41,44 @@ const paginate = <T>(users: T[], options: PaginationOptions): T[] => {
   return users.slice(startIndex, endIndex);
 };
 
+const searchSubstringInArray = (searchTerm: string, users: Record<string, User>[]): Record<string, User>[] => {
+  const results = new Set<Record<string, User>>();
+
+  users.filter((userObj) =>
+    Object.values(userObj).some((user) => {
+      Object.values(user).forEach((value) => {
+        if (typeof value === 'string' && value.toLowerCase().includes(searchTerm?.toLowerCase())) {
+          results.add(userObj);
+          return;
+        }
+      });
+    }),
+  );
+
+  return Array.from(results);
+};
+
+const createUser = <V extends User>(data: Partial<V>): V => {
+  const newUser = {
+    id: '',
+    firstName: '',
+    lastName: '',
+    email: '',
+  } as V;
+
+  for (const key in data) {
+    if (key in newUser) {
+      newUser[key as keyof V] = data[key] as V[keyof V];
+    }
+  }
+  return newUser;
+};
+
+const userRole = <K extends string, V extends User>(role: K, user: V): Record<string, V> => {
```

Comment:
maybe `createUserRole` could be a better name for the function?

Reply:
changed